### PR TITLE
buildChunkGraph code clarification

### PR DESCRIPTION
**Pull request by goganchic**
For now it's a bit difficult to understand code of buildChunkGraph. I'm trying to introduce minor code/comments modifications for better understanding.

**What kind of change does this PR introduce?**

refactoring

**Did you add tests for your changes?**

no

**Does this PR introduce a breaking change?**

no

**What needs to be documented once your changes are merged?**

nothing

### CODE REVIEW

File: lib/buildChunkGraph.js [hunk 1]
```diff
@@ -277,7 +277,7 @@ const visitModules = (
 	 * @returns {void}
 	 */
 	const iteratorBlock = b => {
-		// 1. We create a chunk for this Block
+		// 1. We create a chunk group with single chunk in it for this Block
```
Comment (goganchic) [inline lib/buildChunkGraph.js]: Is my comment correct?

File: lib/buildChunkGraph.js [hunk 2]
```diff
@@ -327,15 +327,16 @@ const visitModules = (
...
+		// for the chunk group which is set in processQueue
+		let deps = chunkGroupDependencies.get(chunkGroup);
+		if (!deps) chunkGroupDependencies.set(chunkGroup, (deps = []));
 		deps.push({
 			block: b,
 			chunkGroup: c
 		});
 
-		// 3. We create/update the chunk group info
+		// 3. We enqueue the cunk group info creation/updating
```
Comment (goganchic) [inline lib/buildChunkGraph.js]: Would someone please give some comments why we need to enqueue chunkGroupInfo modification? Why we can't just modify it here without additional queue?

**Comment by goganchic** [comment]
@sokra Is it correct to say that during graph building there always only one chunk in every chunk group?

File: lib/buildChunkGraph.js (hunk 2 above)
Comment (sokra) [inline lib/buildChunkGraph.js]:
```suggestion
		// 3. We enqueue the chunk group info creation/updating
```

It's enqueued for performance reasons. We group multiple items together to save duplicate work. In addition to that some work need to be done after all modules for the current chunk groups has been walked to get module indices correct.

**Comment by goganchic** [comment]
@sokra you define variable `let queue = [];`, but after that you use it not as a queue, but as a stack (push/pop). Should I rename this variable to `stack`?

**Pull request merged**